pkgs/development/python-modules/numba/default.nix

```diff
@@ -70,14 +70,15 @@ in buildPythonPackage rec {
     done
   '';
 
-  # Copy test script into $out and run the test suite.
+  # run the test suite in a temporary directory so
+  # a) Python picks up the installed library in $out instead of the build files
+  # b) we have somewhere to put $HOME so some caching tests work
   checkPhase = ''
-    ${python.interpreter} -m numba.runtests
+    pushd $(mktemp -d)
```

Comment:
Btw, are these the tests you mentioned are super expensive to compute? If so, we should move them to `passthru.tests` and limit the `checkPhase` to something that is reasonably cheap to run often

Reply:
Yes, they are 30-40 mins in real time on my 12 core system so not easy but not horrendous. We could select some subset of the tests to run there but I don't really know which would be most representative.

Alternately, we could simply rely on the `pythonImportsCheck` like we have been. What would moving to `passthru.tests` look like? Just a `runCommand`?

Reply:
> Just a `runCommand`?

Yes, I think so.

As for the package's `checkPhase`, it seems that we can run just a small subset of tests: https://github.com/numba/numba/blob/60723cb5a31d0b6ef516af95602a463322991420/docs/source/developer/contributing.rst?plain=1#L176

```console
❯ python -m numba.runtests -l numba.tests.test_usecases
numba.tests.test_usecases.TestUsecases.test_andor
numba.tests.test_usecases.TestUsecases.test_blackscholes_cnd
numba.tests.test_usecases.TestUsecases.test_copy_arrays
numba.tests.test_usecases.TestUsecases.test_copy_arrays2d
numba.tests.test_usecases.TestUsecases.test_string_comparisons
numba.tests.test_usecases.TestUsecases.test_string_concat
numba.tests.test_usecases.TestUsecases.test_string_conversion
numba.tests.test_usecases.TestUsecases.test_string_len
numba.tests.test_usecases.TestUsecases.test_string_slicing
numba.tests.test_usecases.TestUsecases.test_sum1d
numba.tests.test_usecases.TestUsecases.test_sum1d_pyobj
numba.tests.test_usecases.TestUsecases.test_sum2d
numba.tests.test_usecases.TestUsecases.test_while_count
13 tests found. 13 selected
❯ time python -m numba.runtests numba.tests.test_usecases
...
Ran 13 tests in 1.567s

OK

real	0m2.055s
user	0m1.979s
sys	0m0.064s
```

We should keep the `pythonImportsCheck` too, of course. These two should be a sufficient heuristic that it's worth to keep a built numba package in the cache...